src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -9413,6 +9413,158 @@ void LF()
             Assert.False(comp.HasLocalsInit("C.<>c.<M>b__0_5")); // LL
         }
 
+        [Fact]
+        public void SkipLocalsInit_LocalFunctionAttribute()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+
+public class C
+{
+    public void M()
+    {
+        [System.Runtime.CompilerServices.SkipLocalsInitAttribute]
+        void F()
```

Comment:
>void F() [](start = 8, length = 8)

I think we should also test scenarios when the attribute is applied to a containing member (including events/properties if allowed), or containing type, or containing module.

Reply:
Some of these scenarios already had tests, such as `SkipLocalsInitAttributeOnMethodPropagatesToLocalFunction`, but I added tests for the other scenarios you mentioned.

---
In reply to: [371418225](https://github.com/dotnet/roslyn/pull/41183#discussion_r371418225) [](ancestors = 371418225)